### DISCUSSION TITLE
Fritzbox 7583 VDSL is working great

### DISCUSSION THREAD

**Discussion by devmarxx** [Freetz-NG/freetz-ng#1046] [Allgemein]
Today I build the image for the 7583 VDSL with OS 7.59.

The building went without any problems, the flashing worked with 

"tools/push_firmware -ram 128 -lfs 9"

Everything is working fine, so from my point of view the remark "UNTESTED" in menuconfig can be removed :-)

Regards

Chris

![7583vdsl](https://github.com/user-attachments/assets/1cadd9b8-3a29-4ec7-9706-d3310fcec798)

**Comment by fda77** [comment]
Danke! Kannst du noch die "AVM-Info"  Seite mit ram und cpu posten?
Ging der default von 208MB nicht (https://github.com/Freetz-NG/freetz-ng/blob/master/tools/push_firmware#L803) und du hast `-ram 128` verwendet?

**Comment by devmarxx** [comment]
Default habe ich "tools/push_firmware" probiert. Das hat nicht funktioniert. Dann habe ich diesen Hinweis gefunden:

https://www.ip-phone-forum.de/threads/7590-unterschied-push-firmware-von-freetz-ng-und-dem-recovery-tool-von-avm.305091/post-2348705

Damit hat es geklappt.

AVM-Info:

![avm-info](https://github.com/user-attachments/assets/18a3dcfb-d752-4ff3-9779-dcba44d5b0cb)

**Comment by fda77** [comment]
Das dürfte eine der schnellsten MIPS Fritzboxen sein, für DSL 50 etwas überdimensioniert ;-)

Ob ncftp oder nicht verwendet wird kann man mit `-cmd` festgelegt werden: https://github.com/Freetz-NG/freetz-ng/blob/master/tools/push_firmware#L494

128MB von den 512MB RAM vorhanden scheint mir aber recht wenig! Dann das neue Image das via ADAM2 hochgeladen wird muss damit booten können und sich selbst flashen. Bei kleinen Images geht das, bei größeren evtl nicht

Am besten würde man ein avm-recovery machen und die ftp-kommunikation mitschneiden (port 21 tcp) und schauen was avm mit memsize nutzt.

Ich wollte das emulieren, aber ich bin entweder schlecht beim Environment raten, oder das avm-recovery ist wirklich kaputt
![image](https://github.com/user-attachments/assets/35a936f8-054b-47fd-a08d-3fe8c6bd538d)

kannst du die (anonymisierte) Ausgabe von `cat /proc/sys/urlader/environment` schicken?

EDIT: es liegt an meinem falschen env, mit einer alten recovery von https://web.archive.org/web/20230715000000*/https://download.avm.de/fritzbox/fritzbox-7583-vdsl/deutschland/recover/ bekomm ich auch crc-fehler

**Reply by devmarxx** [reply]
> Das dürfte eine der schnellsten MIPS Fritzboxen sein, für DSL 50 etwas überdimensioniert ;-)
> 
More power :-))

Hier das "enviroment":


DMC	SL1
HWRevision	260
HWSubRevision	1
ProductID	Fritz_Box_HW260
SerialNumber	P091756000xxxxx
annex	A
autoload	yes
bootloaderVersion	1.11216
country	049
firstfreeaddress	0x852A5A00
firmware_info	260.07.59
firmware_version	avme
flashsize	nor_size=0MB sflash_size=0KB nand_size=512MB
language	de
linux_fs_start	0
maca	xx:xx:xx:xx:xx:xx
macb	xx:xx:xx:xx:xx:xx
macwlan	xx:xx:xx:xx:xx:xx
macwlan2	xx:xx:xx:xx:xx:xx
macdsl	xx:xx:xx:xx:xx:xx
memsize	0x20000000
mtd0	0x0,0x2C00000
mtd1	0x500000,0xD00000
mtd2	0x0,0x100000
mtd3	0x100000,0x500000
mtd4	0xD00000,0x1500000
mtd5	0x1500000,0x20000000
my_ipaddress	192.168.178.1
prompt	Eva_AVM
ptest	
usb_board_mac	xx:xx:xx:xx:xx:xx
usb_device_id	0x0000
usb_device_name	USB DSL Device
usb_manufacturer_name	AVM
usb_revision_id	0x0000
usb_rndis_mac	xx:xx:xx:xx:xx:xx
webgui_pass	xxxxxxxx
wlan_key	xxxxxxxxxxxxxxxxxxxx
wlan_ssid	FRITZ!Box#7583#TI

**Comment by fda77** [comment]
Danke, das war leider nicht das Problem! Es klappt einfach nicht mit dem Emu, vielleicht war es ein VM update oder Windows update oder Java update - keine Ahnug Bleibt leider nur ein ein Dump einer Recovery-Sitzung um den ram zu finden der zugewiesen wird

**Comment by devmarxx** [comment]
Ich kann gerne mal schauen, ob ich das hinkriege. Kannst Du mir sagen, wie so ein Recovery Dump geht?

Ich habe hier alles auf Linux, aber die recovery.exe läuft doch auch unter "wine" meine ich?

**Reply by fda77** [reply]
Wine hab ich noch nie benutzt, musst du ausprobieren. Ein avm-recovery läuft aber mindestens in VMware-VMs ("bridged"), bei mir geht Windows 7 besser als 10 oder 11 wegen dem ganzen Netzwerkkram
Ein Dump ist einfach ein Mittschnitt der ftp-Verbindung, es wird auch nur Port 21 benötigt was ein ein paar Bytes sind (auf 20/tcp wird das Image selbst übertragen). Unter Windows ist Wireshark gut, oder tcpdump wenn ein Linux dazwischen hängt.
Am besten die Fritzos Einstellungen vorher sichern!

**Comment by devmarxx** [comment]
Wow ... Das war ein Spaß.

Ich habe das 7.59 Recovery mittels "wine" einspielen können und dabei mit 

tcpdump einen ASCII Mitschnitt auf Port 21 gemacht. Anbei:

[ftp_dump.txt](https://github.com/user-attachments/files/17228109/ftp_dump.txt)

So far so good, aber dann habe ich 1,5 Stunden rumgefrettet Freetz-NG wieder zurück zu spielen. push_firmware wollte sich par tout nicht mehr verbinden. Nach endlosen esotherischen Versuchen ging es dann plötzlich doch. Wieder mit den Parametern "-ram 128 -lfs 9". Warum auch immer. Ich habe dann die Sicherung zurückgespielt und alles ist wieder back to normal.

Ich hoffe sehr, dass der Dump was bringt. War schwer erkämpft :-))

**Comment by fda77** [comment]
Das ist leider keine FTP Session, ich weiss auch nicht was es ist! Das was man auf Port 21 übertragen wird ist Klartext, man sieht Befehle wie user, pass, type, media oder das gesuchte setzten des rams.
Ich hab aber auch bemerkt dass ab Fritzos 7.5 avm was am recovery geändert hat, mein Emulator geht mit keinem Gerät mehr und alle melden "crc"

**Comment by devmarxx** [comment]
Ich habe den Dump mit 

tcpdump -A -i en0 'port 21'

gemacht. War wohl nicht richtig so ...

Ich bin da jetzt aber erstmal raus. Das Gefrett um Freetz-NG wieder zurück zu bekommen ist mir doch zu deftig.

Wenn es noch jemand probieren möchte ... Was bei mir (wahrscheinlich so) geklappt hat war:

1. Die Fritzbox mit dem origial System hochfahren
2. Den (Linux-) PC mit Netzwerkkabel an LAN1 der Box verbinden (IP des PC dann 192.168.178.2)
3. Auf der Konsole vom PC "tools/push_firmware -ram 128 -lfs 9" starten
4. Fritzbox vom Strom nehmen
5. Fritzbox wieder am Strom hängen

Damit "sollte" das Flashen funktionieren.

**Comment by fda77** [comment]
Ich weiss dass das nervig ist! Also wenn es nicht zu viele Daten sind mache ich einen kompletten Mitschnitt mit tcpdump in eine Datei. Die kann man danach mit Wireshark öffnen und filtern, zb nach Port oder Session (follow tcp stream) und den text herauskopieren

**Comment by devmarxx** [comment]
Es lässt mich irgendwie noch nicht los ...

Da die 7583 VDSL bei mir regulär in Einsatz ist und der Recovery-Vorgang immer zur Folge hat, dass ich Freetz-NG danach wieder neu flashen darf und die Sicherung zurückspielen muss, ist es nicht so prickelnd das solange zu machen, bis ich endlich das Auslesen der RAM-Werte richtig mache mittels tcpdump. Das kann ein langwieriges trial and error werden ...

Jetzt habe ich das hier gefunden:

https://github.com/Freetz-NG/freetz-ng/discussions/923

Kann ich nicht einfach per FTP in dem kurzen Bootfenster (Sekunde 5-7 nach dem Booten) mit Benutzername "adam2" und Passwort "adam2" auf die Box und dann mit "quote GETENV <parameter>" die Box abfragen?

Wenn dem so ist, welche Parameter brauchen wir dann (außer alle :-))?

memsize? mtd0 bis mtd4?

**Comment by devmarxx** [comment]
Nicht fragen, einfach machen :-))

ProductID             Fritz_Box_HW260
memsize               0x20000000
mtd0                  0x0,0x2C00000
mtd1                  0x500000,0xD00000
mtd2                  0x0,0x100000
mtd3                  0x100000,0x500000
mtd4                  0xD00000,0x1500000
mtd5                  0x1500000,0x20000000

**Comment by fda77** [comment]
Das wäre zu einfach! Aber ich kann verstehen dass ein recovery aufwändig ist, vor allem wenn es der Router ist.
Dennoch wird der RAM gesucht der WÄHREND des reovery nutzbar ist! Das wird via FTP festgelegt ("quote memsize").
Du selbst hast den Wert auf "128 mb" gesetzt weil es in irgend einer Anleitung stand (ohne überhaupt zu verstehen was du das festgelegt hast)
Diesen Wert den AVM nutzt ist in der recovery-exe fest drin und man müsste schon ein recovery mitschneiden

**Reply by fda77** [reply]
Ich hab den Parameter mal umbenannt damit das aufhört:
 * 078ab7f38
